mlflow/lightgbm.py

```diff
@@ -73,7 +74,7 @@ def get_default_pip_requirements():
              Calls to :func:`save_model()` and :func:`log_model()` produce a pip environment
              that, at minimum, contains these requirements.
     """
-    return [_get_pinned_requirement("lightgbm")]
+    return [_get_pinned_requirement("lightgbm"), _get_pinned_requirement("cloudpickle")]
```

Comment:
Can we add `cloudpickle` conditionally because users who don't use scikit-learn estimators don't need `cloudpickle`?

Reply:
Hi @harupy, thank you for your suggestion! Does that mean we also need to provide an option to turn on / off autologging for scikit-learn estimators? I assume `mlflow.lightgbm.autolog()` enable autologging for all models.

Reply:
Hi @harupy, I found a simple way to add `cloudpickle` conditionally (and automatically) based on what model is saved (please see [L149-150](https://github.com/mlflow/mlflow/pull/5130/files#diff-37c34422216cf5a04a1ef09cf2095615e62927e03d51819d4c020a57ced7719cR149-R150)). Please let me know your feedback and comments. Thanks a lot!

(This [comment](https://github.com/mlflow/mlflow/pull/5130#discussion_r772765954) is also addressed in the latest commit.)